src/Native/CpuMath/IntrinsicsUtils.cs

```diff
@@ -0,0 +1,796 @@
+                    }
+                }
+            }
+        }
+
+        public static void AddScaleSU(float scale, float[] src, int[] idx, float[] dst, int count)
+        {
+            if (Sse.IsSupported)
+            {
+                AddScaleSUSse(scale, src, idx, dst, count);
+            }
+            else
+            {
+                for (int i = 0; i < count; i++)
+                {
+                    dst[idx[i]] += scale * src[i];
+                }
+            }
+        }
+
+        private static void ScaleUSse(float scale, float[] dst, int count)
+        {
+            Vector128<float> scaleVector = Sse.SetAllVector128(scale);
+
+            unsafe
+            {
+                fixed (float* pdst = dst)
+                {
+                    float* pDstCurrent = pdst;
+                    float* pEnd = pdst + count;
+
+                    while (pDstCurrent + 4 <= pEnd)
+                    {
+                        Vector128<float> dstVector = Sse.LoadVector128(pDstCurrent);
+
+                        dstVector = Sse.Multiply(scaleVector, dstVector);
+                        Sse.Store(pDstCurrent, dstVector);
+
+                        pDstCurrent += 4;
+                    }
+
+                    while (pDstCurrent < pEnd)
+                    {
+                        Vector128<float> dstVector = Sse.LoadScalarVector128(pDstCurrent);
+
+                        dstVector = Sse.MultiplyScalar(scaleVector, dstVector);
+                        Sse.StoreScalar(pDstCurrent, dstVector);
+
+                        pDstCurrent++;
+                    }
+                }
+            }
+        }
+
+        public static void ScaleU(float scale, float[] dst, int count)
+        {
+            if (Sse.IsSupported)
+            {
+                ScaleUSse(scale, dst, count);
+            }
+            else
+            {
+                for (int i = 0; i < count; i++)
+                {
+                    dst[i] *= scale;
+                }
+            }
+        }
+
+        private static float Dist2Sse(float[] src, float[] dst, int count)
+        {
+            Vector128<float> SqDistanceVector = Sse.SetZeroVector128();
+
+            unsafe
+            {
+                fixed (float* psrc = src)
+                fixed (float* pdst = dst)
+                {
+                    float* pSrcCurrent = psrc;
+                    float* pDstCurrent = pdst;
+                    float* pEnd = psrc + count;
+
+                    while (pSrcCurrent + 4 <= pEnd)
+                    {
+                        Vector128<float> distanceVector = Sse.Subtract(Sse.LoadVector128(pSrcCurrent), Sse.LoadVector128(pDstCurrent));
+                        SqDistanceVector = Sse.Add(SqDistanceVector, Sse.Multiply(distanceVector, distanceVector));
+
+                        pSrcCurrent += 4;
+                        pDstCurrent += 4;
+                    }
+
+                    // TODO: Check code length per line
```

Comment:
What does this mean - the characters on the line? I don't think we have a firm rule. Use judgement. No need to check in this comment.

Reply:
Perhaps it means something else though?

Reply:
Thanks!  Good to know - I thought I had to make every line less than 79 characters long for the sake of the reader as in those systems assignment.  Just now fixed the code length per line anyway :)  Thank you!

Sorry I should have omitted this 'src\Native\CpuMath` folder from the PR - it was my initial working repo with commented lines, unfixed styles, and TODOs for my development.  Please feel free to ignore this folder, and the codes in `Microsoft.ML.*` are those that get finally integrated into the dotnet repo.  Sorry for any inconvenience caused, and thank you for your understanding.